js/app/modules/selection/subsetsForSet.js

```diff
@@ -31,15 +43,25 @@ const SubsetsForSet = new Module.Class({
             if (this.track_subset) {
                 property_name = 'current-subset';
             }
-            this.model = HistoryStore.get_default()[property_name.replace('-', '_', 'g')];
+            this.model = this._get_model(property_name);
             HistoryStore.get_default().connect('notify::' + property_name, () => {
-                this.model = HistoryStore.get_default()[property_name.replace('-', '_', 'g')];
+                let model = this._get_model(property_name);
+                if (!model || (this.model && model.ekn_id === this.model.ekn_id))
+                    return;
+                this.model = model;
                 this._set_needs_refresh(true);
             });
         }
         this._set_needs_refresh(true);
     },
 
+    _get_model: function (property_name) {
+        let model = HistoryStore.get_default()[property_name.replace('-', '_', 'g')];
+        if (this.track_parent)
```

Comment:
looks like, something this can get a bit racy:

```
(com.endlessm.prensa_libre.es_GT:24458): Gjs-WARNING **: JS ERROR: TypeError: subset is null
get_parent_set@resource:///com/endlessm/knowledge/js/app/setMap.js:29
SubsetsForSet<._get_model@resource:///com/endlessm/knowledge/js/app/modules/selection/subsetsForSet.js:61
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
SubsetsForSet<._init/<@resource:///com/endlessm/knowledge/js/app/modules/selection/subsetsForSet.js:48
HistoryStore<._update_index@resource:///com/endlessm/knowledge/js/app/historyStore.js:208
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
HistoryStore<.go_back@resource:///com/endlessm/knowledge/js/app/historyStore.js:226
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
HistoryStore<._init/<@resource:///com/endlessm/knowledge/js/app/historyStore.js:97
Dispatcher<._process_queue@resource:///com/endlessm/knowledge/js/app/dispatcher.js:53
wrapper@resource:///org/gnome/gjs/modules/lang.js:178
@<command line>:10
```

Means that HistoryStore is returning a null or undefined model, so better check for that as well... this only happened to me twice.